### PR TITLE
[SPARK-49553][PYTHON][DOCS] Remove the experimental API notes for pandas related functions

### PR DESCRIPTION
**Pull request by allisonwang-db**
### What changes were proposed in this pull request?

This PR removes the experimental API notes for applyInPandas/applyInArrow/mapInPandas/mapInArrow.

### Why are the changes needed?

These functions are no longer experimental.

### Does this PR introduce _any_ user-facing change?

No 

### How was this patch tested?

Existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: python/pyspark/sql/pandas/group_ops.py [hunk 1]
```diff
@@ -329,8 +327,6 @@ def applyInPandasWithState(
         Notes
         -----
         This function requires a full shuffle.
-
-        This API is experimental.
```
Comment (allisonwang-db) [inline python/pyspark/sql/pandas/group_ops.py:333]: cc @HeartSaVioR can we remove this for streaming ones?

File: python/pyspark/sql/pandas/group_ops.py [hunk 2]
```diff
@@ -636,8 +630,6 @@ def applyInArrow(
         into memory, so the user should be aware of the potential OOM risk if data is skewed
         and certain groups are too large to fit in memory.
 
-        This API is unstable, and for developers.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/pandas/group_ops.py:639]: Let's keep this for `...InArrow` ones

File: python/pyspark/sql/pandas/group_ops.py (hunk 1 above)
Comment (HeartSaVioR) [reply python/pyspark/sql/pandas/group_ops.py:333]: Yeah good to remove as we passed through one minor release (3.5) and there is no outstanding issue.

File: python/pyspark/sql/pandas/group_ops.py (hunk 2 above)
Comment (allisonwang-db) [reply python/pyspark/sql/pandas/group_ops.py:639]: `mapInArrow` should be stable right? This applyInArrow is indeed a new API.

**Review by HyukjinKwon — APPROVED** [review]

**Review by xinrong-meng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**